packages/components/src/palette-edit/index.tsx

```diff
@@ -475,10 +568,14 @@ export default function PaletteEdit( {
 			{ hasElements && (
 				<>
 					{ isEditing && (
-						<PaletteEditListView
+						<PaletteEditListView< typeof elements[ number ] >
 							canOnlyChangeValues={ canOnlyChangeValues }
 							elements={ elements }
-							onChange={ onChange }
+							onChange={
+								onChange as (
+									newElement?: Array< Color | Gradient >
+								) => void
```

Comment:
This is hacky. 

<details>
<summary>TS error it prevents</summary>

```sh
packages/components/src/palette-edit/index.tsx:574:8 - error TS2322: Type '((newColors?: Color[] | undefined) => void) | ((newGradients?: Gradient[] | undefined) => void)' is not assignable to type '(newElements?: (Color | Gradient)[] | undefined) => void'.
  Type '(newColors?: Color[] | undefined) => void' is not assignable to type '(newElements?: (Color | Gradient)[] | undefined) => void'.
    Types of parameters 'newColors' and 'newElements' are incompatible.
      Type '(Color | Gradient)[] | undefined' is not assignable to type 'Color[] | undefined'.
        Type '(Color | Gradient)[]' is not assignable to type 'Color[]'.
          Type 'Color | Gradient' is not assignable to type 'Color'.
            Property 'color' is missing in type 'Gradient' but required in type 'Color'.

574        onChange={
           ~~~~~~~~

  packages/components/src/palette-edit/types.ts:12:2
    12  color: string;
        ~~~~~
    'color' is declared here.
  packages/components/src/palette-edit/types.ts:118:2
    118  onChange: ( newElements?: T[] ) => void;
         ~~~~~~~~
    The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & PaletteEditListViewProps<Color | Gradient>'


Found 1 error.
```
</details